src/dataflow/src/render/reduce.rs

```diff
@@ -1859,11 +1867,14 @@ pub mod monoids {
                         lhs.clone_from(&rhs);
                     }
                 }
-                (lhs, rhs) => log::error!(
-                    "Mismatched monoid variants in reduction! lhs: {:?} rhs: {:?}",
-                    lhs,
-                    rhs
-                ),
+                (lhs, rhs) => {
+                    assert_or_log!(
```

Comment:
this seems like it wants to be `panic_or_log!`? (nbd if you don't want to add that)